File: backend_v2/src/main/java/at/htlleonding/pepper/entity/dto/StepDto.java
package at.htlleonding.pepper.entity.dto;

import at.htlleonding.pepper.entity.Game;
import at.htlleonding.pepper.entity.Image;
import at.htlleonding.pepper.entity.Move;
import at.htlleonding.pepper.entity.Step;

public record StepDto(Game game, int index, String image, String image_desc, Move move, String text, int durationInSeconds) {
}

File: pep-caretaker/PepperKotlin_Frontend/mmg/app/src/main/java/com/example/mmg/dto/StepDto.kt
package com.example.mmg.dto

import com.google.gson.annotations.SerializedName

data class StepDto(
    @SerializedName("index") val index: Int,
    @SerializedName("move") val move: MoveDto? = null,
    @SerializedName("text") val text: String,
    @SerializedName("durationInSeconds") val durationInSeconds: Int,
    @SerializedName("image") val image: ImageDto? = null
)

data class MoveDto(
    @SerializedName("id") val id: Int,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String
)

data class ImageDto(
    @SerializedName("description")val description: String,
    @SerializedName("id")val id: Int,
)